infrastructure/haproxy_reload.sh

```diff
@@ -21,20 +22,38 @@ preconfigure() {
   iptables -w -t nat -N HAPROXY
   iptables -w -t nat -A PREROUTING -j HAPROXY
   iptables -w -t nat -A OUTPUT -j HAPROXY
+  
   for chain in haproxy_a haproxy_b; do
     instance_prefix="${chain}_prefix"
-    real_port="${!instance_prefix}${real}"
+    http_port="${!instance_prefix}${http}"
     stats_port="${!instance_prefix}${stats}"
     iptables -w -t nat -N ${chain}
-    [ ${LISTEN_IP} != "0.0.0.0" ] && \
-    iptables -w -t nat -A ${chain} -p tcp -d ${LISTEN_IP} --dport 80 -j DNAT --to-destination ${LISTEN_IP}:${real_port}
-    iptables -w -t nat -A ${chain} -m state --state NEW -p tcp -d ${HOST_IP} --dport 80 -j REDIRECT --to ${real_port}
+    [[ ${LISTEN_IP} != "0.0.0.0" ]] && {
+       [[ ${HAPROXY_SSL} == "false" ]] && {
+		   iptables -w -t nat -A ${chain} -p tcp -d ${LISTEN_IP} --dport 80 -j DNAT --to-destination ${LISTEN_IP}:${http_port}
+       } || {
+    	   iptables -w -t nat -A ${chain} -p tcp -d ${LISTEN_IP} --dport 443 -j DNAT --to-destination ${LISTEN_IP}:${http_port} 
+       }   
+    }
+    [[ ${HAPROXY_SSL} == "false" ]] && {
+   		iptables -w -t nat -A ${chain} -m state --state NEW -p tcp -d ${HOST_IP} --dport 80 -j REDIRECT --to ${http_port}
+
+    } || {
+   		iptables -w -t nat -A ${chain} -m state --state NEW -p tcp -d ${HOST_IP} --dport 443 -j REDIRECT --to ${http_port}    	    
+    }
+    
```

Comment:
Instead of doing doubble iptables rules just do:
```
DPORT=80
 [ ${HAPROXY_SSL} == "true" ] && DPORT=443
```
and the use `--dport $DPORT`
here below the same